pages/docs/scale/docker_swarm.html

```diff
@@ -0,0 +1,172 @@
+
+```bash
+gcloud compute instances create machine-master \
+  --network default \
+  --zone us-central1-a \
+  --boot-disk-size 50GB \
+  --image centos-7 \
+  --machine-type f1-micro \
+  --metadata-from-file startup-script=startup-master.sh
+```
+
+Save the Project ID as an environment variable to avoid repetition:
+
+```bash
+echo "export GCE_PROJECT=PROJECT_ID" >> ~/.bash_profile
+source ~/.bash_profile
+```
+
+### Obtain Cluster Token
+
+Create a Swarm token, a unique token used to identify the Swarm cluster.
+
+Create a remote Docker host first:
+
+```bash
+docker-machine create -d google --google-project ${GCE_PROJECT} swarm-master
+```
+
+Switch to the machine and run `swarm create` which will start a Swarm and the hash of the container is used as the token.
+
+```bash
+eval $(docker-machine env swarm-master)
+docker run swarm create
+```
+
+Note the token for further use or put the token into bash profile, so it's easily available.
+
+```bash
+echo "export TOKEN=TOKEN_VALUE" >> ~/.bash_profile
+source ~/.bash_profile
+```
+
+### Creating Master
+
+The Swarm master is the first instance of the cluster, and all other nodes will connect to that. The master also acts as a regular node in the cluster, meaning that Docker containers can be launched on it.
+
+This time we'll use the `google` driver, `n1-standard-2` instances and a bigger boot disk for a bit more computing power.
+
+The option `--swarm-master` is the key here:
+
+```bash
+docker-machine create -d google --google-project=${GCE_PROJECT} --google-machine-type=n1-standard-2 --google-disk-size=50 --swarm --swarm-master --swarm-discovery token://${TOKEN} crate-swarm
+```
+
+### Create Swarm Nodes
+
+Next create 3 machines for the Swarm nodes, replacing `crate-swarm-node1` with `crate-swarm-node2` etc.
+
+```bash
+docker-machine create -d google --google-project=${GCE_PROJECT} --google-machine-type=n1-standard-2 --google-disk-size=50 --swarm --swarm-discovery token://${TOKEN} crate-swarm-node1
+```
+
+### Creating a Crate Cluster
+
+Now we want to start multiple Crate nodes that form a cluster. Unfortunately, Google Cloud Platform (and other cloud hosting network environments) do not have multicast enabled, so Crate needs to use another method for discovering other Crate nodes.
+
+There are a variety of options including:
+
+- Unicast discovery
+- [DNS discovery](/docs/reference/en/latest/configuration.html#conf-discovery)
+- Using a discovery backend, like [Consul](https://www.consul.io/)
+
+The simplest is unicast discovery, but it's not the most dynamic. As this documentation is more about Swarm, we will use Unicast discovery.
+
+Because Crate is running inside a container, the Crate node will need to publish the IP or hostname of its host. Usually you know on which host you're running your Crate container, so can pass the hostname in the `docker run` command. When using Swarm you don't know where your container will be deployed when you execute the `docker run` command.
```

Comment:
@chaudum I was going to update this and then realised that we still need the bash script steps to get the list of hosts, so I'll just remove the line.